src/test/e2e/api/admin/segment.e2e.test.ts

```diff
@@ -108,17 +108,19 @@ const validateSegment = (
         .expect(expectStatusCode);
 
 beforeAll(async () => {
-    db = await dbInit('segments_api_serial', getLogger);
-    app = await setupAppWithCustomConfig(
-        db.stores,
-        {
-            experimental: {
-                flags: {
-                    anonymiseEventLog: true,
-                    detectSegmentUsageInChangeRequests: true,
-                },
+    const customOptions = {
+        experimental: {
+            flags: {
+                anonymiseEventLog: true,
+                detectSegmentUsageInChangeRequests: true,
             },
         },
+    };
+
+    db = await dbInit('segments_api_serial', getLogger, customOptions);
```

Comment:
If not made aware of the custom options, any flag resolution in the stores will fail.